### PR TITLE
[Build] Upgrade build to Gradle 7.2 and improve build

### PR DESCRIPTION
**Pull request by lhotari**
### Motivation

Upgrade build to Gradle 7.2 and use io.github.lhotari.gradle-nar-plugin

### Modifications

- update plugin versions
- use io.github.lhotari.gradle-nar-plugin
- make build run without specifying maven credentials

### Open question

- remove odd solution from jar task since it's no more possible to use this type of code in Gradle 7.2
```groovy
jar {
    zip64=true
    from {
        configurations.compile.collect { it.isDirectory() ? it : zipTree(it) }
    }
}
```
- is there a reason to make the jar behave like shadowJar? This seems confusing.

### CODE REVIEW

File: producer-v3-pulsar/build.gradle [hunk 1]
```diff
@@ -8,14 +8,13 @@ application {
     mainClassName = "$mainClassName"
 }
 
+configurations.implementation.canBeResolved = true
```
Comment (lhotari) [inline producer-v3-pulsar/build.gradle:11]: @vroyer this should be removed, since that shouldn't be needed.

File: producer-v3/build.gradle [hunk 2]
```diff
@@ -6,16 +6,17 @@ application {
     mainClassName = "$mainClassName"
 }
 
+configurations.implementation.canBeResolved = true
+configurations.compileOnly.canBeResolved = true
```
Comment (lhotari) [inline producer-v3/build.gradle:10]: @vroyer this should be removed, since that shouldn't be needed.

Comment (lhotari) [reply producer-v3/build.gradle]: I removed these.

**Comment by vroyer** [comment]
Cherry-picked in the master branch

**Pull request closed without merging**